src/core/controller/models/refreshOcaModels.ts

```
@@ -0,0 +1,135 @@
+import { StringRequest } from "@shared/proto/cline/common"
+import { OcaCompatibleModelInfo, OcaModelInfo } from "@shared/proto/cline/models"
+import axios from "axios"
+import { HostProvider } from "@/hosts/host-provider"
+import { OcaAuthService } from "@/services/auth/oca/OcaAuthService"
+import { DEFAULT_OCA_BASE_URL } from "@/services/auth/oca/utils/constants"
+import { createOcaHeaders, getProxyAgents } from "@/services/auth/oca/utils/utils"
+import { Logger } from "@/services/logging/Logger"
+import { ShowMessageType } from "@/shared/proto/index.host"
+import { Controller } from ".."
+
+/**
+ * Refreshes the Oca models and returns the updated model list
+ * @param controller The controller instance
+ * @param request Empty request object
+ * @returns Response containing the Oca models
+ */
+export async function refreshOcaModels(controller: Controller, request: StringRequest): Promise<OcaCompatibleModelInfo> {
+	const parsePrice = (price: any) => {
+		if (price) {
+			return parseFloat(price) * 1_000_000
+		}
+		return undefined
+	}
+	const models: Record<string, OcaModelInfo> = {}
+	let defaultModelId: string | undefined
+	const ocaAccessToken = await OcaAuthService.getInstance().getAuthToken()
```

Check for missing token here and fail fast with informative error via HostProvider.window.showMessage